cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -581,38 +631,38 @@ private void closeScanner() {
   }
 
   /**
-   * Filters elements matching an index {@link ScannerIterator}.
+   * Filters elements matching a field {@link ScannerIterator}.
    */
   @VisibleForTesting
-  static final class FilterByIndexIterator extends AbstractCloseableIterator<StructuredRow> {
+  static final class FilterByFieldIterator extends AbstractCloseableIterator<StructuredRow> {
     private final CloseableIterator<StructuredRow> scannerIterator;
     private final Predicate<StructuredRow> predicate;
 
-    FilterByIndexIterator(CloseableIterator<StructuredRow> scannerIterator, Field<?> filterIndex,
+    FilterByFieldIterator(CloseableIterator<StructuredRow> scannerIterator, Field<?> filterField,
```

Comment:
why are we changing this name?

Reply:
Because we can relax it to not just index fields, we could filter by any fields, right?

Reply:
is the sql implementation doing that?

Reply:
no, it's noSQL that we're doing post-filtering

Reply:
So we do check if it is an index in the nosql implementation to be consistent with the spi: https://github.com/cdapio/cdap/blob/develop/cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java#L206

Reply:
so i think it should be `FilterByFieldIterator` to be consistent

Reply:
also such changes - if they don't affect the intended PR changes- it is better to create a separate one?

Reply:
Yeah I'll update the spi as well.

Reply:
sorry I meant `FilterByIndexIterator` but, lets discuss offline in a bit

Reply:
discussed offline, resolving this